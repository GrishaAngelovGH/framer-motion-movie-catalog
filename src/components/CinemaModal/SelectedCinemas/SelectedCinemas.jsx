const SelectedCinemas = ({ cinemas, onRemove }) => (
  <div className="m-5 max-h-[150px] md:max-h-full overflow-auto">
    {
      cinemas.map(v => (
        <span key={v.id} onClick={() => { onRemove(v.id) }} className="bg-gray-200 border border-gray-400 rounded-full m-2 p-2 inline-flex items-center hover:cursor-pointer">
          <span>{v.name}</span>
          <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="size-4 inline">
            <path strokeLinecap="round" strokeLinejoin="round" d="M6 18 18 6M6 6l12 12" />
          </svg>
        </span>
      ))
    }
  </div>
)

export default SelectedCinemas