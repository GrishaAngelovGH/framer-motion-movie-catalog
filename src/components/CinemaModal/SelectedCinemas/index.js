import SelectedCinemas from "./SelectedCinemas"

export default SelectedCinemas