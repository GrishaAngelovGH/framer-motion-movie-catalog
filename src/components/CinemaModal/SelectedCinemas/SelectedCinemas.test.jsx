import { render } from "@testing-library/react"

import SelectedCinemas from "./SelectedCinemas"

test("should render SelectedCinemas component", () => {
  const view = render(
    <SelectedCinemas cinemas={[
      { id: "8498625877351675", name: "London Leicester Square", address: "22-24 Leicester Square, London" }
    ]} />
  )

  expect(view).toMatchSnapshot()
})