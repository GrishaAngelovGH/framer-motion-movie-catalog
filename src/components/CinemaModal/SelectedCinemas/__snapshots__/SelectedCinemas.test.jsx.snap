// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render SelectedCinemas component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-5 max-h-[150px] md:max-h-full overflow-auto"
      >
        <span
          class="bg-gray-200 border border-gray-400 rounded-full m-2 p-2 inline-flex items-center hover:cursor-pointer"
        >
          <span>
            London Leicester Square
          </span>
          <svg
            class="size-4 inline"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18 18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-5 max-h-[150px] md:max-h-full overflow-auto"
    >
      <span
        class="bg-gray-200 border border-gray-400 rounded-full m-2 p-2 inline-flex items-center hover:cursor-pointer"
      >
        <span>
          London Leicester Square
        </span>
        <svg
          class="size-4 inline"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18 18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
